.github/workflows/release-cycle.yml

```
@@ -0,0 +1,34 @@
+name: Test release cycle
+
+on: [pull_request, push, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Silence Windows warnings
+        run: |
+          git config core.eol lf
+          git config core.autocrlf true
```

```suggestion
          git config core.autocrlf input
```

This *should* fix the test, to my knowledge and also per local testing

---

Will contributors / release managers using windows run into the same problem, and be required to have this same config?

---

Good question; to make a long story short, the answer is in fact no; this particular issue is essentially a "false positive" in the test we're using on CI. Read on for the detailed explanation...

----

The problem here was that because `core.autocrlf` was set to `true`, Git converted the LF EOLs to CRLF when the files were checked out on Windows. Then, when the script ran, it output the files with a consistent LF rather than the platform default, per my previous review suggestion. This causes the files in the working directory to appear as if they were modified. However, if `core.autocrlf` is set to `true`, Git will conform them back to LF when they are staged to the index, so the EOL won't actually change in the index or the repo. To note, if the test had done `git stage .` first, the test would have (correctly) passed.

Therefore, if Windows users have `core.autocrlf` set to `true`, the files will show as modified in the working tree, but once they do `git add .`, they will be conformed and no changes will be shown or committed (we _could_ prevent the minor display nit by reading the existing files, checking the EoL and outputing with that, but IMO the extra complexity that increases the chance of an _actual_ bug isn't worth it). If they have it set to `input` or `false`, the files will be checked out as LF, so there will be no changes at any point, since the script always outputs LF.

To note, there _would_ be an actual problem if the script used the default platform-specific newline output, and a Windows user had `core.autocrlf` set to `false` instead of `true` or `input`, but that was fixed in an earlier round of review by always using `\n` on output.

---

Still fails with `git config core.autocrlf input`:
```
HEAD detached at pull/988/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   include/branches.csv
	modified:   include/end-of-life.csv
	modified:   include/release-cycle.mmd

no changes added to commit (use "git add" and/or "git commit -a")
```
https://github.com/python/devguide/actions/runs/3580813722/jobs/6023265680

> The problem here was that because core.autocrlf was set to true, Git converted the LF EOLs to CRLF when the files were checked out on Windows. 

Aha, we need to set the config _before_ the checkout! And with `--global` because we don't have a Git repo yet.

[`a87279d` (#988)](https://github.com/python/devguide/pull/988/commits/a87279dc382c040a828e7a044e52b930b6b9c35f)

https://github.com/hugovk/devguide/actions/runs/3580902248/jobs/6023437583

:tada: Thanks!

---

Yeah, I'd realized that as well once the CI had failed, but was in the process of recommending a simpler and more robust solution (staging the files first) based on this discussion.